### PR TITLE
Emoji in code block, fix issue #329

### PR DESCRIPTION
**Pull request by jollypolly123**

### CODE REVIEW

File: functions/src/content.ts [hunk 1]
```diff
@@ -70,7 +70,13 @@ function replaceEmojis(md: string, emojis: Record<string, string>): string {
   for (const k of Object.keys(emojis)) {
     const withColons = `:${k}:`;
     const asEmoji = emojis[k];
-    content = content.replace(new RegExp(withColons, "g"), asEmoji);
+    const contentList = content.split("```");
```
[the remark this answers is not in the document]
Comment (jollypolly123) [reply functions/src/content.ts]: Got it—just wondering, why wouldn't we want to do the first change?

[the remark this answers is not in the document]
Comment (samtstern) [reply functions/src/content.ts]: @jollypolly123 based on my reading of [the spec](https://github.github.com/gfm/#fenced-code-blocks) I think the triple backticks have to be the start of a line or preceded only by whitespace on the line in order to fence off a code block. So it is possible (although unlikely) that you could have a "```" in the middle of content. Therefore just splitting on that character group is not enough.

Simple:
```
Hello
```

Triple backticks at end of line:
```
Hello```
```

Indented
  ```foo
Ok
  ```

File: functions/src/content.ts [hunk 2]
```diff
@@ -66,14 +66,23 @@ export function renderContent(
...
-  let content = md;
+  let content = md.split("\n");
+  let isInCodeBlock = false;
+  
   for (const k of Object.keys(emojis)) {
     const withColons = `:${k}:`;
     const asEmoji = emojis[k];
-    content = content.replace(new RegExp(withColons, "g"), asEmoji);
+    content.forEach((val, ind) => {
+      if (val.includes("```")) {
```
Comment (samtstern) [inline functions/src/content.ts:76]:
```suggestion
      if (val.trim().startsWith("```")) {
```

**Review by samtstern — CHANGES_REQUESTED** [review]
Just one more change, thanks!

File: functions/src/content.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jollypolly123) [reply functions/src/content.ts]: Gotcha thanks!

**Review by samtstern — APPROVED** [review]

**Comment by samtstern** [comment]
@jollypolly123 thank you! This should be fixed the next time @techtrailhead deploys the site.

**Pull request merged**